caf: Add ble_state (#4180)
nrfconnect/sdk-nrf: include/caf/ble_state.rst

.. _caf_ble_state:

CAF: Bluetooth LE state module
##############################

.. contents::
   :local:
   :depth: 2

> The Bluetooth LE state module is responsible for:

Reviewer: ```suggestion
When enabled for an application, the |ble_state| is responsible for the following actions:
```

* Enabling Bluetooth (:c:func:`bt_enable`).
* Handling Zephyr connection callbacks (:c:struct:`bt_conn_cb`).
* Propagating information about the connection state and parameters by using :ref:`event_manager` events.

Configuration
*************

To use the module, you must enable the following Kconfig options:

* :option:`CONFIG_BT`.

> * :option:`CONFIG_BT_SMP` - The module requires at least connection security level 2 (encryption).

Reviewer: ```suggestion
* :option:`CONFIG_BT_SMP` - The module requires at least the connection security level 2 (encryption).
```

> * :option:`CONFIG_CAF_BLE_STATE` - This option enables the Bluetooth LE state module and selects the :option:`CONFIG_CAF_BLE_COMMON_EVENTS` that enables Bluetooth LE common events in CAF.

Reviewer: ```suggestion
* :option:`CONFIG_CAF_BLE_STATE` - This option enables the Bluetooth LE state module and selects the :option:`CONFIG_CAF_BLE_COMMON_EVENTS` Kconfig option, which enables Bluetooth LE common events in CAF.
```

The following Kconfig options are available for this module:

> * :option:`CONFIG_CAF_BLE_STATE_EXCHANGE_MTU`
>   This option can be used for GATT client (:option:`CONFIG_BT_GATT_CLIENT`) to set the MTU to the maximum possible size the buffers can hold.
>   This option is enabled by default.
> * :option:`CONFIG_CAF_BLE_USE_LLPM`
>   This option enables the Low Latency Packet Mode (LLPM).
>   This option depends on :option:`CONFIG_BT_CTLR_LLPM` and :option:`CONFIG_BT_LL_SOFTDEVICE` and it's enabled by default.

Reviewer: ```suggestion

* :option:`CONFIG_CAF_BLE_STATE_EXCHANGE_MTU` - This option can be used for GATT client  :option:`CONFIG_BT_GATT_CLIENT`) to set the MTU to the maximum possible size the buffers can hold.
  This option is enabled by default.
* :option:`CONFIG_CAF_BLE_USE_LLPM` - This option enables the Low Latency Packet Mode (LLPM).
  This option depends on :option:`CONFIG_BT_CTLR_LLPM` and :option:`CONFIG_BT_LL_SOFTDEVICE` and it's enabled by default.
```

Implementation details
**********************

> The |ble_state| module is used by both Bluetooth Peripheral and Bluetooth Central devices.

Reviewer: ```suggestion
The |ble_state| is used by both Bluetooth Peripheral and Bluetooth Central devices.
```
I guess only those that are using CAF?

Author: Yes, like for all other CAF modules.
